pyvista/core/dataset.py

```diff
@@ -45,6 +45,8 @@
     from collections.abc import Generator
     from collections.abc import Iterator
 
+    from typing_extensions import Self
```

Comment:
I see this now is for backwards compatibility.  Should we be testing for backwards compatibility?  I started this #6980